dotnet/src/SemanticKernel.Abstractions/AI/FunctionChoiceBehaviors/FunctionChoiceBehaviorConfiguration.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Microsoft.SemanticKernel;
+
+/// <summary>
+/// Represent function choice behavior configuration.
+/// </summary>
+[Experimental("SKEXP0001")]
+public sealed class FunctionChoiceBehaviorConfiguration
```

Comment:
Should this contain an abstraction for the tool choice property?

How can I use tool choice none and still pass a list of functions?

Reply:
> Should this contain an abstraction for the tool choice property?

The idea was to allow connectors use the choice behavior classes polymorphically via the parent class - `FunctionChoiceBehavior` so the connectors would not even be aware which specific choice behavior classes they are working with. Adding the property would go against the polymorphic nature of the solution and that can be fine if the solution doesn't allow us to achieve what we want. Or you are thinking putting an abstraction over the choice property?

> How can I use tool choice none and still pass a list of functions?

Advertising functions to model for the "none" choice were not originally planned. None of the documents I saw imply passing both:
* OpenAI - https://platform.openai.com/docs/guides/function-calling
   <img width="972" alt="image" src="https://github.com/microsoft/semantic-kernel/assets/68852919/fe88c9be-e8c4-483f-98f1-c59664b5f61d">
* Gemini - https://cloud.google.com/vertex-ai/generative-ai/docs/multimodal/function-calling
   <img width="628" alt="image" src="https://github.com/microsoft/semantic-kernel/assets/68852919/ced2d7c7-f681-4edd-96d0-9ac72f825b23">
* Mistral - https://docs.mistral.ai/capabilities/function_calling/
   <img width="190" alt="image" src="https://github.com/microsoft/semantic-kernel/assets/68852919/224abd6e-0503-4431-9ae5-812c8b607b14">

However, I recognize that the model, ideally, should support it in case a caller wants to supply both.

Reply:
fixed